rust/frontend/src/function.rs

```diff
@@ -195,11 +195,11 @@ impl<'a, 'm> Builder<'a, 'm> {
         let (mut values, mut type_codes): (Vec<ffi::TVMValue>, Vec<ffi::TVMTypeCode>) = self
             .arg_buf
             .iter()
-            .map(|tvm_arg| (tvm_arg.value, tvm_arg.type_code as ffi::TVMTypeCode))
+            .map(|arg| arg.clone().into_tvm_value())
```

Comment:
Can it be without `clone` after `into_iter` or something?

Reply:
that would require

```diff
- type PackedFunc = Fn(&[TVMArgValue]) -> TVMRetValue;
+ type PackedFunc = Fn(Vec<TVMArgValue>) -> TVMRetValue;
```

and the TVM module is surely not going to know how to Drop a Vec (nor can the frontend or runtime Drop a C++ std::vector).

I think that a better solution might be `.as_tvm_value()`, but that adds a lot  of boilerplate code and it's not totally clear that rustc doesn't just optimize away the clone.

Reply:
Well, `as_tvm_value` seems correct. I guess I'll just make the macros more macro-y to eliminate the repetitive code.

Reply:
You could eliminate the allocation on the rust side doing something like:

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=8d4048c7235d7e7bf8074f1514a07350

This wouldn't be able to go through `Builder` though.

Also, mapping a slice of tuples to a pair of slices is actually surprisingly hard to do with macros. You could avoid doing it if you're able to just call `into()` twice for each macro, or else you can do some [advanced macro magic](https://danielkeep.github.io/tlborm/book/pat-push-down-accumulation.html) to map over the slice.

Reply:
This can be addressed later.